logserver/src/main/java/ai/vespa/logserver/protocol/ArchiveLogMessagesMethod.java

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package ai.vespa.logserver.protocol;
+
+import com.yahoo.jrt.DataValue;
+import com.yahoo.jrt.Int32Value;
+import com.yahoo.jrt.Int8Value;
+import com.yahoo.jrt.Method;
+import com.yahoo.jrt.Request;
+import com.yahoo.logserver.LogDispatcher;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * RPC method that archives incoming log messages
+ *
+ * @author bjorncs
+ */
+class ArchiveLogMessagesMethod {
+
+    static final String METHOD_NAME = "vespa.logserver.archiveLogMessages";
+
+    private static final Logger log = Logger.getLogger(ArchiveLogMessagesMethod.class.getName());
+
+    private final Executor executor = Executors.newSingleThreadExecutor();
+    private final LogDispatcher logDispatcher;
+    private final Method method;
+
+    ArchiveLogMessagesMethod(LogDispatcher logDispatcher) {
+        this.logDispatcher = logDispatcher;
+        this.method = new Method(METHOD_NAME, "bix", "bix", this::log)
+                .methodDesc("Archive log messages")
+                .paramDesc(0, "compressionType", "Compression type (0=raw)")
+                .paramDesc(1, "uncompressedSize", "Uncompressed size (0 if compression type 'raw')")
```

Comment:
The code further down requires uncompressedSize == 'data length', so the comment should be updated.

Reply:
Fixed